src/filetree.rs

```diff
@@ -280,12 +280,63 @@ pub(crate) fn syncfs(d: &openat::Dir) -> Result<()> {
     rustix::fs::syncfs(d).map_err(Into::into)
 }
 
+fn copy_dir(src: &Path, dst: &Path) -> Result<()> {
+    let r = std::process::Command::new("cp")
```

Comment:
To make this operate in terms of relative paths, the standard approach I use looks like this:
https://github.com/coreos/bootupd/blob/6a4fc6f8758ff18432d5a43995b8f53ad79d4419/src/filesystem.rs#L33

I think using that we may be able to drop the usage of `recover_path`?